contracts/MechMarketplace.sol

```diff
@@ -365,6 +364,29 @@ contract MechMarketplace is IErrorsMarketplace {
         emit SetMechFactoryStatuses(mechFactories, statuses);
     }
 
+    // Check and escrow delivery rate
+    function _checkAndEscrowDeliveryRate(address mech) internal {
```

Comment:
We hold whatever incoming funds on MM (escrow) relevant contracts based on a `maxDeliveryRate`.